src/main/java/core/basesyntax/RightTriangle.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax;
+
+public class RightTriangle extends Figure {
+    private double base;
+    private double height;
+    private double hypotenuse;
+
+    public RightTriangle(Color color, double base, double height) {
+        super(color);
+        this.base = base;
+        this.height = height;
+        this.hypotenuse = Math.sqrt(base * base + height * height);
+    }
+
+    @Override
+    public double getArea() {
+        return 0.5 * base * height;
+    }
+
+    @Override
+    public void print() {
+        System.out.println("Figure: right triangle, area: " + getArea()
+                + " sq. units, base: " + base
+                + " units, height: " + height + " units, hypotenuse: " + hypotenuse
+                + " units, color: " + new ColorSupplier().getRandomColor());

```

Comment:
Avoid creating a new instance of 'ColorSupplier' inside the 'print' method. It's better to pass the color as a parameter to the constructor of the 'RightTriangle' and use it directly.